apps/condo/domains/resident/access/Resident.js

```diff
@@ -18,8 +19,24 @@ async function canReadResidents ({ authentication: { item: user } }) {
 async function canManageResidents ({ authentication: { item: user }, originalInput, operation, itemId, context }) {
     if (!user) return false
     if (user.isAdmin) return true
-    if (operation === 'create' && user.type === RESIDENT) {
-        return true
+    if (user.type === RESIDENT) {
+        if (operation === 'create') return true
+        // Only soft-delete is allowed for current resident
+        if (operation === 'update') {
```

Comment:
It’s better to check only deletedAt ticket

Reply:
You also should check if the deletedAt is the only key in the query

Right now I can update some properties while deleting an object

Reply:
Added extra checks in `737cbaa1`